systems/subscriber/sim-pool/pkg/server/sim.go

```diff
@@ -0,0 +1,132 @@
+package server
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	"github.com/ukama/ukama/systems/common/grpc"
+	mb "github.com/ukama/ukama/systems/common/msgBusServiceClient"
+	"github.com/ukama/ukama/systems/common/msgbus"
+	pb "github.com/ukama/ukama/systems/subscriber/sim-pool/pb/gen"
+	"github.com/ukama/ukama/systems/subscriber/sim-pool/pkg"
+	"github.com/ukama/ukama/systems/subscriber/sim-pool/pkg/db"
+	"github.com/ukama/ukama/systems/subscriber/sim-pool/pkg/utils"
+)
+
+type SimPoolServer struct {
+	simRepo        db.SimRepo
+	msgbus         mb.MsgBusServiceClient
+	baseRoutingKey msgbus.RoutingKeyBuilder
+	pb.UnimplementedSimServiceServer
+}
+
+
+func NewSimPoolServer(simRepo db.SimRepo, msgBus mb.MsgBusServiceClient) *SimPoolServer {
+	return &SimPoolServer{simRepo: simRepo,
+		msgbus:         msgBus,
+		baseRoutingKey: msgbus.NewRoutingKeyBuilder().SetCloudSource().SetContainer(pkg.ServiceName)}
+}
+
+func (p *SimPoolServer) Get(ctx context.Context, req *pb.GetRequest) (*pb.GetResponse, error) {
```

Comment:
Is this suppose to return list or one single non-allocated sim ?

Reply:
A single sim

Reply:
Implementation says otherwise